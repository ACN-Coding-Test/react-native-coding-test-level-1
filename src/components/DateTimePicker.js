import React from "react";
import PropTypes from "prop-types";
import { StyleSheet } from "react-native";
import RNDateTimePicker from "@react-native-community/datetimepicker";
import theme from "../theme";

function DateTimePicker(props) {
  const { value, inputStyles, ...otherProps } = props;
  return (
    <RNDateTimePicker
      mode="date"
      value={value}
      style={[styles.inputStyles, inputStyles]}
      {...otherProps}
    />
  );
}

DateTimePicker.defaultProps = {
  value: new Date(),
};

DateTimePicker.propTypes = {
  value: PropTypes.any,
  inputStyles: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.object,
    PropTypes.array,
  ]),
};

const styles = StyleSheet.create({
  inputStyles: {
    width: "100%",
    fontSize: theme.fontSizes.body,
    padding: theme.padding.default,
    borderRadius: theme.borderRadius.default,
  },
});

export default DateTimePicker;
