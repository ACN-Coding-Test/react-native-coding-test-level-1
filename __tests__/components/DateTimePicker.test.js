import React from 'react'
import DateTimePicker from '../../src/components/DateTimePicker'
import { render, waitFor } from '@testing-library/react-native';


const renderPicker = async (props) => {
  const utils = render(<DateTimePicker value={new Date()} {...props} />);
  await waitFor(() => utils.UNSAFE_getByType(DateTimePicker));
  return utils;
};


describe('DateTimePicker:', () => {
  it('renders a native Component', async () => {
    const { toJSON } = await renderPicker();
    expect(toJSON()).toMatchSnapshot();
  });
})
