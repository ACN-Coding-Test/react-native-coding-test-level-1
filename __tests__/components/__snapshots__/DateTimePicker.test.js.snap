// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker: renders a native Component 1`] = `null`;

exports[`DateTimePicker: should render correctly 1`] = `null`;
